libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -16,6 +17,12 @@ AP_Camera_Backend::AP_Camera_Backend(AP_Camera &frontend, AP_Camera_Params &para
 // update - should be called at 50hz
 void AP_Camera_Backend::update()
 {
+    if (_params.options == 1 && AP::dal().get_armed()) {
+        this->record_video(true);
```

Comment:
no need for "this->"

Reply:
also should be using the dal.  The dal is really only for use by the EKF.  There's a HAL soft_armed() method that can be used instead.